File: sdks/csharp/src/NovaBilling/Invoices/Requests/MarkPaidInvoicesRequest.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record MarkPaidInvoicesRequest
{
    /// <summary>
    /// Invoice ID
    /// </summary>
    [JsonIgnore]
    public required string Id { get; set; }

    /// <summary>
    /// Payment method used (cash, bank_transfer, check, manual). Defaults to "manual".
    /// </summary>
    [JsonPropertyName("paymentMethod")]
    public string? PaymentMethod { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/src/Invoices/Requests/MarkPaidInvoicesRequest.php
<?php

namespace NovaBilling\Invoices\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;

class MarkPaidInvoicesRequest extends JsonSerializableType
{
    /**
     * @var ?string $paymentMethod Payment method used (cash, bank_transfer, check, manual). Defaults to "manual".
     */
    #[JsonProperty('paymentMethod')]
    public ?string $paymentMethod;

    /**
     * @param array{
     *   paymentMethod?: ?string,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->paymentMethod = $values['paymentMethod'] ?? null;
    }
}
